### PR TITLE
Improves Estimize Downloaders

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

* Add completion tracking in Estimize downloaders (excluding consensus)
* Add better logging to estimize downloaders
* Speed up estimize consensus downloader by checking for duplicates
* Move duplicated code to single method
#### Related Issue

#3404 
#### Motivation and Context

More support in LEAN for alternative data sources
#### Requires Documentation Change

No
#### How Has This Been Tested?

Tested locally on Windows 10
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by gsalaz98** [comment]
This is not yet ready for merging or review - I'm currently implementing mapping for tickers and discovered another issue similar to the one fixed in the TradingEconomics downloader:

https://github.com/QuantConnect/Lean/blob/8e256e202a68588aa4d9552b79a09b4fe0a0bda0/ToolBox/TradingEconomicsDataDownloader/TradingEconomicsCalendarDownloader.cs#L100-L112

Will have a fix out soon once I've finished verifying that there's no issues with the data with the new transformations we will apply

**Comment by gsalaz98** [comment]
Ready for review 👍

File: Common/Data/Custom/Estimize/EstimizeEstimate.cs [hunk 1]
```diff
@@ -98,6 +99,33 @@ public class EstimizeEstimate : BaseData
...
+        public EstimizeEstimate(string csvLine)
+        {
+            // CreatedAt[0], Id[1], AnalystId[2], UserName[3], FiscalYear[4], FiscalQuarter[5], Eps[6], Revenue[7], Flagged[8]"
+            var csv = csvLine.Split(',');
+
+            CreatedAt = DateTime.ParseExact(csv[0], "yyyyMMdd HH:mm:ss", CultureInfo.InvariantCulture);
+            Id = csv[1];
+            AnalystId = csv[2];
+            UserName = csv[3];
+            FiscalYear = Convert.ToInt32(csv[4], CultureInfo.InvariantCulture);
```
Comment (jaredbroad) [inline Common/Data/Custom/Estimize/EstimizeEstimate.cs]: Nice use of invariant

**Comment by gsalaz98** [comment]
RateGate threw an error while downloading the data. Perhaps a(n) issue(s) should be opened as it pertains to this user's comment in: http://www.jackleitch.net/2010/10/better-rate-limiting-with-dot-net/

```
Jim says	

November 20, 2013 at 7:36 pm

Jack, I’ve noticed that the implementation seems suffer from two flaws:
First: The implementation of the ExitTimerCallback checks the queue during a while() loop, but then again at a subsequent if(). This is unnecessary, but also this if()’s TryPeek() output may be past-due, resulting in a negative number in timeUntilNextCheck.

Second: Environment.TickCount rolls over every 24.9 days, which can also mess with constant-up service deployments.
```

I've added extra logic to prevent RateGate failures from tanking the download process should it fail

**Pull request merged**